packages/astro/src/core/sync/index.ts

```diff
@@ -124,6 +124,14 @@ export async function syncInternal({
 			});
 			await contentLayer.sync();
 			settings.timer.end('Sync content layer');
+		} else if (fs.existsSync(getContentPaths(settings.config, fs).contentDir.href)) {
```

Comment:
Does this work on Windows? Using `href` might be unsafe.

Reply:
It was required to make some tests pass (see first commit). I manually tested it and it seems to work